### DISCUSSION TITLE
저희는 ABC-USER-FEEDBACK를 활용중입니다.

### DISCUSSION THREAD

**Discussion by Magga-Lee** [line/abc-user-feedback#1018] [Ideas]
### Describe the feature you'd like to request

안녕하세요.

우선 ABC-USER-FEEDBACK에 관해 Tech Blog와 Git Hub의 소스를 보았을때 많은 시간과, 다양한 고민, 그리고 기술의 노하우가 담겨있는 프로젝트임을 알았고 그런 기술과 노하우를 공개해주신것에 대해 진심으로 경의와 감사를 표합니다.

저는 GS리테일에 근무중이고, GS리테일에서는 사내 임직원의 브랜드 서비스의 사용자 피드백(버그 리포트)를 받는 용도로 ABC-USER-FEEDBACK을 활용해 보고 있습니다.

저희의 대략적인 피드백의 흐름은 아래와 같습니다.
1. 사내 임직원이 모바일 서비스를 이용중에 임직원들에게만 보이는 버그 신고 버튼을 통해 서비스 화면에 버그 / 기능 제안 내용을 작성.
2. ABC-USER-FEEDBACK API를 통해 작성 내용 수집.
3. ABC-USER-FEEDBACK ADMIN을 통해 각 영역 담당자들이 피드에 대한 내용 확인후 확인 내용에 대한 EDIT항목 입력 및 관리.
   ( Edit항목 : 피드유형 , 담당자정보등) 

다양한 곳에서 활발하게 ABC-USER-FEEDBACK활용되고 지속적인 발전을 위해 사용하면서 아쉬웠던 부분이 있어 의견을 공유드려 봅니다.

1. 필드 유형에 Link 타입이 있으면 어떨까요? 저희는 수집하는 항목에 이미지 주소 혹은 페이지 주소가 있습니다.
   (이미지의 경우 S3등을 연동해 image 필드를 통해 직접 이미지를 관리 할수도 있으나, 저희처럼 주소만 수집하는 경우, 필요에 따라 URL주소를 열어 보기 위해 피드에서 URL을 긁어 브라우저에 복사/붙여넣기를 하여 확인하고 있습니다.

2. edit 항목은 리스트 상에서만 입력가능하기 때문에 내용이 많아 상세보기 열어 본 후 입력하기 위해 상세를 닫고 다시 편집모드로 전환후 입력항목으로 이동해야하는 불편함이 있습니다. 그렇게 때문에 처음부터 편집모드로 변경후 리스트상에 피드 확인후 필드 입력을 하거나 상세보기전 읽을 피드가 어떤건지 잘 기억하고 있어야합니다. 상세보기 화면에서도 edit모드 전환 기능이 있었으면 좋겠습니다.

3. 2번의 경우가 불편한 이유는 가로스크롤 처리에 있는것 같습니다.
가로 스크롤이 리스트 마지막에 표시됩니다. 피드 항목이 많은 경우 50/100개 리스트 보기로 화면을 볼경우 가로 스크롤이 화면하단에 생기는게 아니라 리스트 페이지 하단에 생기기 때문에 리스트 제일 마지막 번호 까지 내려가서 가로 스크롤 이동을 해야합니다.
저희는 22개의 피드항목을 가지고 있고 이중 8개는 피드에 대한 피드백 관리용으로 edit기능을 부여하여 abc-user-feedback에서 직접 내용을 기입하는데 왼쪽에서 부터 수집된 유저 피드백 내용을 보고 오른쪽에 확인 내용을 입력하는 형태를 취하고 있습니다. 이렇다 보니 해상도에 따라 가로 스크롤에 대한 불편함이 있습니다.

4. Edit의 Text형에 멀티 라인 입력 지원이 가능하면 좋겠습니다. 
2번의 상세화면에서의 에디트 모드가 있었으면 좋게 다는 의견은 Text필드에 대한 입력의 불편함이 있습니다. 리스트 상에 편집모드로 텍스트 필드에 장문의 문장을 넣기에 작성글을 다 볼수 없어 불편함이 있습니다. 담당자들은 장문 작성을 메모장등의 텍스트 에디터를 이용하여 내용먼저 작성하고 작성한 내용을 붙여넣기하여 등록하고 있습니다.

5. 수집된 이슈/피드백외에 admin에서도 직접 피드와 이슈 등록이 가능하면 다양한 곳에서 abc-user-feedback 을 활용 할수 있을것 같습니다.
경우에 따라 abc-user-feedback만으로 권한을 부여받은 사용자끼리 이슈를 생성하고, 피드를 작성하고 싶은 경우도 있지 않을까 합니다.

 6. 대시보드도 약간의 에디트 기능이 있으면 대시보드 활용이 좀더 다양하게 이루어질 수 있을것 같습니다.
예를 들어 SELECT형의 특정 필드에 대한 집계기능을 추가 할수 있거나, 키워드 필드형 집계를 추가 할 수 있거나 혹은 데시보드 항목중 비노출 하고 싶은 항목들을 편집할수 있다면 좀더 대시보드를 활용에 도움이 될것 같습니다.
 
 7. 동일 이슈에 대해 묶여 있는 사용자 피드의 개별적인 상태관리.
저희같은 경우 "로그인"이라는 이슈를 만들고, 관련된 피드를 받게 되면 이슈관리 화면에서 로그인 이슈에 하위로 수집된 로그인 피드들이 쌓여 보기게 됩니다.

이슈에 주제는 동일하나 하위에 달려있는 피드들은 각기 다른 내용으로 개별적인 피드에 관리가 필요한데, 이슈관리 화면에서는 이슈에 달린 피드를 모두 묶어서 상태관리해야 하는것 같습니다. 

로그인에 관련된 새로운 피드는 지속적으로 수집되기 때문에 이슈화면에서 상태관리의 기능이 역할이 모호 하게 됩니다. 저희가 활용중인 서비스로 보면 저희 같은경우 지금 구조에서는 사용자가 피드를 생성하게 되면 먼저 이슈를 신규 유형으로 만들고 사용자가 등록한 내용을 이슈에 하위로 달아야 하는데 그렇게 되면 "로그인"이라는 한가지 이슈 주제로 관리하는게 아니라 "로그인-0001"등의 유니크한 이슈 유형을 만들어야 하는걸로 보여집니다. 로그인이란 이슈 주제에 대해 수집되는 서로 다른 다양한 피드에 대해 개별적인 상태관리도 있다면 좋을것 같습니다.

다시 한번 감사드립니다.

### Describe the solution you'd like to see

https://github.com/line/abc-user-feedback//categories/ideas

**Comment by YoungjaeKim** [comment]
안녕하세요! 친절히 피드백 주셔서 감사합니다.
팀에서 7개 항목에 대한 답을 준비하여 곧 댓글로 말씀드리겠습니다. 🙇

**Comment by YoungjaeKim** [comment]
@Magga-Lee 안녕하세요. 답이 늦어서 실례했습니다.
주신 질문에 따라 아래에 답을 붙이겠습니다. 결론부터 말씀드리자면 1월 중에 다음 주 버전을 릴리스할 예정인데, 이 때 몇 가지는 해소될 것으로 보입니다.

1. Link 타입 - 타입을 추가하기보다는 텍스트 중에 하이퍼링크 형식에 대해 클릭할 수 있도록 규칙(regex)을 적용하겠습니다. 이 경우 보안적으로 injection의 위험이 있는데요, 이것은 의견을 입력받는 쪽에서 처리해서 걸러주는게 좋겠다고 생각합니다. 왜냐하면 저희 프로덕트(AUF) 자체적으로 입력값을 필터링했다간 injection 규칙은 꾸준히 바뀌기에 임의로 삭제시킬 수 있기 때문입니다.

2. Edit 접근 경로 개선 - 2025.1 릴리즈 예정입니다.

3. 가로 스크롤 - 위의 Edit 접근 경로를 개선하면 사용성이 올라가서 어느정도 해소되지 않을까 기대합니다. 한편, 이미 아실 수도 있습니다만 목록 우상단에 컬럼 정렬과 숨김 기능이 있습니다. 자주 수정하는 컬럼은 위치를 왼쪽으로 조정해보시면 좀 더 나을 수 있습니다.

4. 멀티라인 입력 - Edit 창에는 멀티라인으로 입력할 수 있는데요, 그래서 이 역시 2번의 Edit 접근 경로가 개선되면 어느정도 답답함이 해소될 것으로 기대합니다.

5. 이슈/피드백의 Admin 입력 - 기존에는 한 개의 피드백이라도 있어야 이슈를 생성할 수 있었습니다. 이에 대해, 피드백이 없어도 이슈를 생성할 수 있도록 개선할 예정입니다. 한편, 피드백을 Admin이 임의로 입력할 수 있는 것은 저희가 모르는 시나리오가 있는 것 같습니다. 사용 시나리오를 말씀주시면 고민해보겠습니다. 기존에는 테스트용 글쓰기 정도로 설정 화면에서 만들까 생각한 적은 있습니다.

6. 대시보드의 SELECT를 활용한 커스텀 - 현재는 말씀대로 프로젝트 단위로 다소 기본적인 수준의 지표만 보여주고 있습니다. 그리고 팀에서는 말씀하신 점에 대해 꽤 공감을 했습니다. 그래서 가능하면 1분기 안에 준비할 수 있도록 하겠습니다.

7. 이슈 그룹 - 이슈를 상하위 구조로 만들 수 있는 것에 대해서는 꽤 오래 얘기를 했습니다. 일단 저희 팀 안에서의 사용성도 분명 이에 대한 요구가 있습니다. 데이터 구조적으로는 어려움이 적은데, UX가 좀 고민됩니다. 상위 이슈라는 표시를 어떻게 표시해야 덜 헷갈릴지 아직 답이 없는데요, 기능에 대해서는 공감하고 있으므로, 최대한 준비해보도록 하겠습니다. 이것은 고민이 좀 더 필요해보입니다.

이 정도로 팀에서 정리했는데요. 1월의 릴리즈를 경험해보신 후 또 적극적으로 피드백 주시면 다음 개발 로드맵에 큰 도움이 됩니다. 🥇 

좋은 피드백 감사합니다!

**Comment by Magga-Lee** [comment]
안녕하세요. 

자세한 답변 정말 감사합니다.

예정된 릴리즈 버전에 불편했던 점이 개선될 예정이라니 너무도 반가운 소식입니다.

해당 릴리즈 버전이 출시 되면 저희쪽에 반영작업을 통하여 내부 사용자들의 반응을 공유드릴 수 있도록 하겠습니다.

감사합니다.

**Comment by YoungjaeKim** [comment]
@Magga-Lee 안녕하세요. 오랫만에 이 글에 다시 왔습니다.

이번 [v7 릴리즈](https://github.com/line/abc-user-feedback/releases)에 UI/UX를 큰 폭으로 개선하여, 처음 주신 의견의 많은 부분이 해소되었기에 하나씩 다시 짚어보며 답을 드리고자 합니다.

1. OK. `Text` 필드를 추가하여, URL 형태의 경우 링크로 나타납니다.
2. OK. 이제 피드백을 클릭하면 우측에 오버레이 패널로 내용이 나옵니다. 이 패널에서 직접 수정도 가능합니다.
3. OK. 2번의 오버레이 패널 UI에서 수정할 수 있게 되어서, 수정할 때의 스크롤 문제 또한 해소되었습니다.
4. OK. `Text` 타입은 이제 멀티라인을 지원하고, 여러 줄일 경우 스크롤도 나옵니다. 참고로 한 줄 짜리는 `Keyword` 타입이 있습니다.
5. OK. `Issue` 메뉴에 `Create Issue` 버튼을 추가했습니다.
6. 대시보드는 이번에 디자인만 개편했고, 대시보드 수정 기능은 아직 못만들어서 백로그에 넣었습니다. 올해 가을 즈음 만들 것 같습니다.
7. OK. `Issue Category` 라는 개념을 만들었습니다. 하나의 `Issue Category` 안에 여러 `Issue`를 넣을 수 있습니다.

꼭 v7으로 업데이트 해보시길 권해드리며, 이번 개편에 주신 피드백이 큰 도움이 되었습니다.

사용해보신 후 언제라도 또 조언 부탁드립니다.
본 건은 대시보드 외에는 모두 수정된 것 같으므로, 일단 닫도록 하겠습니다.

감사합니다.
